backend/blueprints/spa_api/utils/query_params_handler.py

```diff
@@ -75,17 +74,39 @@ def parse_query_params(query_params: List[QueryParam], args: Dict[str, str], add
 def create_validation_for_query_params(query_params: List[QueryParam], provided_params: List[str]):
     if provided_params is None:
         provided_params = []
-    check_dict:Dict[str, List[str]] = dict()
+    check_dict: Dict[str, List[str]] = dict()
+    list_check: Dict[str, List[QueryParam]] = dict()
     for query in query_params:
         if query.required_siblings is not None and query.name not in provided_params:
             check_dict[query.name] = query.required_siblings
 
+    for query in query_params:
+        if query.required_siblings is not None and query.name not in provided_params and query.is_list:
+            for sibling in query.required_siblings:
+                if sibling in list_check:
+                    continue
+                for sibling_query in query_params:
+                    if sibling_query == query:
+                        continue
+                    if sibling_query.name == sibling and sibling_query.is_list:
+                        if query.name in list_check:
+                            list_check[query.name].append(sibling_query)
+                        else:
+                            list_check[query.name] = [sibling_query]
+
     def validate(created_query_params):
```

Comment:
The return type of this function should be typed as `Optional[CalculatedError]` I think. 
Would make the correct usage of raise and not return (such as in the change on line 33 of spa_api/utils/decorators.py) clearer.